core/src/eval/operation.rs

```diff
@@ -1853,18 +1853,27 @@ impl<R: ImportResolver, C: Cache> VirtualMachine<R, C> {
 
                                         let twin_index = ctrs_right_sieve
                                             .iter()
-                                            .filter_map(|ctr| ctr.as_ref())
-                                            .position(|other_ctr| {
+                                            // Keep track of the index for later deletion
+                                            .enumerate()
+                                            // Filter out contracts already deduplicated
+                                            .filter_map(|(i, other_ctr)| Some((i, other_ctr.as_ref()?)))
+                                            .find_map(|(i, other_ctr)| {
```

Comment:
Would it be simpler to have `ctrs_right_sieve.iter().position(|maybe_other_ctr| maybe_other_ctr.map_or(false, |other_ctr| /* check contract equality */))`. The point being to fix the indexing by checking the `Some` inside the predicate instead of filtering it out.

Reply:
It's still quite a bit of nesting, but I agree it's using less iterators functions in total, so let's do this :+1: implemented in the latest HEAD